snapcraft/plugins/roscore.py

```diff
@@ -67,6 +68,8 @@ def build(self):
 
         os.chmod(service_wrapper, 0o755)
 
+        self._finish_build()
```

Comment:
Well that's sort of the point though-- there are two generic functions, _prepare_build() and _finish_build(). The names indicate when they run, not what they do (since what they do is kind of a smattering of things).